driver-core/src/main/com/mongodb/client/model/expressions/StringExpression.java

```diff
@@ -35,13 +37,22 @@ public interface StringExpression extends Expression {
 
     StringExpression substr(IntegerExpression start, IntegerExpression length);
 
-    default StringExpression substr(int start, int length) {
+    default StringExpression substr(final int start, final int length) {
         return this.substr(of(start), of(length));
     }
 
     StringExpression substrBytes(IntegerExpression start, IntegerExpression length);
 
-    default StringExpression substrBytes(int start, int length) {
+    default StringExpression substrBytes(final int start, final int length) {
         return this.substrBytes(of(start), of(length));
     }
+
+    IntegerExpression parseInteger();
```

Comment:
Because toString and parse pairs are supposed to be able to convert to and from the same value (for example, `false` should toString to `"false"` which should parse back to `false`, but convert converts it to `true` instead).

Reply:
More generally, are you arguing that much (perhaps most) of the functionality of `$convert` should not be exposed in this API?

Reply:
Right.

There are some conversions that are unexpected, or "broken" in the sense that

`valueA.eq( valueA.toTypeB().toTypeA() )` is false, even for values that have a reasonable representation across both types.

Copying from a comment:

One might expect to see all conversions in `$convert` represented in this API, but we expose only the useful ones.

Useful conversions that should be exposed in our API:
- anything-as-string: toString to a parsable type (excludes doc, array)
- parse-string-to-anything: every type should allow parsing to a string
     - presently excludes objects (docs) and arrays
     - includes parsing date strings with a specific format
- integer milliseconds since epoch to date, and vice-versa

Convert also defines many conversions that do not seem useful for our API
- boolean-number conversions: t/f to-from 1/0, which is 
- boolean-other (oid/str/date) - always returns true, this is broken for strings
- number-number - this changes the "underlying" json type, which is not visible in our API; it is better to express such an operation using something like `round`/`floor`/etc.